[2025-10-18T03:11:50.270+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: agricultural_sri_annual_report.data_collection.fetch_weather manual__2025-10-18T02:31:24.144743+00:00 [queued]>
[2025-10-18T03:11:50.275+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: agricultural_sri_annual_report.data_collection.fetch_weather manual__2025-10-18T02:31:24.144743+00:00 [queued]>
[2025-10-18T03:11:50.275+0000] {taskinstance.py:1361} INFO - Starting attempt 4 of 6
[2025-10-18T03:11:50.280+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): data_collection.fetch_weather> on 2025-10-18 02:31:24.144743+00:00
[2025-10-18T03:11:50.282+0000] {standard_task_runner.py:57} INFO - Started process 67 to run task
[2025-10-18T03:11:50.284+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'agricultural_sri_annual_report', 'data_collection.fetch_weather', 'manual__2025-10-18T02:31:24.144743+00:00', '--job-id', '12', '--raw', '--subdir', 'DAGS_FOLDER/agricultural_sri_pipeline.py', '--cfg-path', '/tmp/tmpsx6vzzmt']
[2025-10-18T03:11:50.285+0000] {standard_task_runner.py:85} INFO - Job 12: Subtask data_collection.fetch_weather
[2025-10-18T03:11:50.305+0000] {task_command.py:416} INFO - Running <TaskInstance: agricultural_sri_annual_report.data_collection.fetch_weather manual__2025-10-18T02:31:24.144743+00:00 [running]> on host 4e2b74bc69c8
[2025-10-18T03:11:50.336+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='alerts@agcompany.com' AIRFLOW_CTX_DAG_OWNER='agricultural_risk_team' AIRFLOW_CTX_DAG_ID='agricultural_sri_annual_report' AIRFLOW_CTX_TASK_ID='data_collection.fetch_weather' AIRFLOW_CTX_EXECUTION_DATE='2025-10-18T02:31:24.144743+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-10-18T02:31:24.144743+00:00'
[2025-10-18T03:11:50.533+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.10/site-packages/***/utils/context.py:314: AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
  warnings.warn(_create_deprecation_warning(k, replacements))

[2025-10-18T03:11:50.534+0000] {agricultural_sri_pipeline.py:129} INFO - üå¶Ô∏è Fetching weather data for 2025
[2025-10-18T03:11:50.534+0000] {weather_data_collector.py:169} INFO - üå§Ô∏è  Fetching weather data for 2025
[2025-10-18T03:11:50.534+0000] {weather_data_collector.py:176} ERROR - ‚ùå No Visual Crossing API key provided
[2025-10-18T03:11:50.540+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/agricultural_sri_pipeline.py", line 137, in fetch_weather_data
    context['ti'].xcom_push(key='weather_states', value=result['states'])
KeyError: 'states'
[2025-10-18T03:11:50.544+0000] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=agricultural_sri_annual_report, task_id=data_collection.fetch_weather, execution_date=20251018T023124, start_date=20251018T031150, end_date=20251018T031150
[2025-10-18T03:11:50.549+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 12 for task data_collection.fetch_weather ('states'; 67)
[2025-10-18T03:11:50.558+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-10-18T03:11:50.571+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
